### PR TITLE
Design system updates

### PR DESCRIPTION
**Pull request by markdalgleish**
This PR contains the design system updates that were done as part of #2642. Now that @jxom is also contributing to the design system, it makes sense to decouple these changes from the feature work so that this can be merged without waiting on a QA pass.

As noted in the other PR, this contains the following updates:

- New `Divider` component.
- New `DebugLayout` for highlighting component boundaries during development.
- Space props are now optional across all layout components.
- `Bleed` now supports the `space` prop for consistency.
- `Inset` now supports `top`/`bottom`/`left`/`right` props.
- `MarkdownText` now supports custom colors and custom link handlers.
- For clarity, the `space`/`nestedSpace` props on `MarkdownText` have been renamed to `paragraphSpace`/`listSpace`.
- `Text` now supports a size of `"23px"`.
- The space scale now supports 1–6px for more fine-tuned spacing.
- `BackgroundProvider` now supports a `style` prop to improve low-level style memoization.
- The design system playground now has a rudimentary color mode toggle.
- Fixed a bug where changing the color mode caused children of `BackgroundProvider` to remount.
- `Stack`, `Row` and `Inline` components have had the direction of their spacing inverted so that it doesn't cover hit targets.

### CODE REVIEW

File: src/design-system/README.md [hunk 1]
```diff
@@ -295,6 +295,22 @@ export default () => (
 );
 ```
 
+#### DebugLayout
```
Comment (jxom) [inline src/design-system/README.md:298]: Starting to feel like this readme is redundant now given we now have the web cheat sheet. Should we point the readme to it instead?

**Review by jxom — APPROVED** [review]
lgtm!

**Review by skylarbarrera — APPROVED** [review]
🪄

File: src/design-system/README.md (hunk 1 above)
Comment (markdalgleish) [reply src/design-system/README.md:298]: Yeah, good point. Update it to only include instructions on running the docs locally?

Comment (markdalgleish) [reply src/design-system/README.md]: I've pushed an update, let me know what you think.

**Comment by jxom** [comment]
looks dope

**Pull request merged**